homeassistant/components/iotawatt/sensor.py

```diff
@@ -169,6 +169,8 @@ def __init__(
             self._attr_unique_id = (
                 f"{data.hub_mac_address}-input-{data.getChannel()}-{data.getUnit()}"
             )
+        elif data.getType() == "Output":
```

Comment:
This isn't what the PR titles states.

Also, this wasn't missing by accident, please check earlier PR about this and all the related controversies

Reply:
I have updated the PR description and title.  I forgot there were 2 commits in my code - one which corrects missing "outputs" sensors not being created in HA since last Fall.  The other was for rates.

Reply:
If you have those unique id for the output sensors, I can guarantee you that your PR will never be merged 

See PR #86834

Reply:
> If you have those unique id for the output sensors, I can guarantee you that your PR will never be merged
> 
> See PR #86834

Why is that format allowed for the inputs then?  Its exactly the same.  The outputs really need to be functional or it defeats the entire point of having the integration if the data can't be pulled in.  Been having to use the HACS version (which is now broken due to other changes and seems un-maintained) because the built in one no longer works with the IoTaWatt outputs.

Not understanding how this is any different than the unique IDs in other things like ESPHome, the IoTaWatt inputs, and a fair number of others?

They are unique values, which are not user-defined (its a combination of the device-specific hardware address and the unique key for the output identifier)...I'm not sure how much more unique it could get?

This makes no sense.

Reply:
It's not the same. Input 1 will always be input 1. It’s unique.

Reply:
But I agree. It makes no sense :)

Reply:
I totally get a need for globally unique identifiers, but also ultimately its only possible to work with what the device itself provides for "primary unique keys".  I suppose one could "invent" intermediate keys in some format but ultimately that would still have to map back to whatever source value and now is back to the original issue of "but there can be {0..n} possible unique keys"

I'm also a bit curious now how it would handle the data-types for an input changing since you can reconfigure it to use some current-sensing inputs for voltage-sensing in order to support multiphase power systems...if its assuming power input 1 will forever be power input 1, that's not a valid assumption either because it might one day change to be voltage input 2 at that input and now you'd have a units mismatch potentially which I have already learned with other stuff REALLY screws up HA if you keep the same entity ID and change the units.  I wonder if that could be a bug that simply hasn't been common enough to have been found/reported yet.